pkg/languagedetection/privileged_detector.go

```diff
@@ -0,0 +1,48 @@
+package languagedetection
+
+import (
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/internal/detectors"
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var detectorsWithPrivilege = []Detector{
+	detectors.GoDetector{},
+}
+
+var (
+	PermissionDeniedWarningOnce = sync.Once{}
+)
+
+func handleDetectorError(err error) {
+	if os.IsPermission(err) {
+		PermissionDeniedWarningOnce.Do(func() {
+			log.Warnf("Attempted to detect language but permission was denied. Make sure the system probe is running as root and has CAP_PTRACE if it is running in a container.")
+		})
+	}
+}
+
+func DetectWithPrivileges(pids []int) []languagemodels.Language {
+	languages := make([]languagemodels.Language, len(pids))
+	for i, pid := range pids {
+		for _, detector := range detectorsWithPrivilege {
+			lang, err := detector.DetectLanguage(pid)
+			if err != nil {
+				handleDetectorError(err)
+				continue
+			}
+			languages[i] = lang
```

Comment:
assume we have 3 pid
we're able to detect the first and last, but we fail for the second
the returns array will be `[real lang, empty, real lang]`
it that what we aim for?

Reply:
That's correct. The default value for `languagemodels.Langauge` is equivalent to:

```go
languagemodels.Language{Name: languagemodels.Unknown}
```

So by leaving a value empty we're effectively saying that we don't know the language